controller/konnect/ops/ops_konnecttransitgateway.go

```diff
@@ -0,0 +1,246 @@
+package ops
+
+import (
+	"context"
+	"fmt"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	sdkkonnectops "github.com/Kong/sdk-konnect-go/models/operations"
+	"github.com/Kong/sdk-konnect-go/retry"
+	"github.com/samber/lo"
+
+	sdkops "github.com/kong/gateway-operator/controller/konnect/ops/sdk"
+
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+// createKonnectTransitGateway creates a transit gateway on the Konnect side.
+func createKonnectTransitGateway(
+	ctx context.Context,
+	sdk sdkops.TransitGatewaysSDK,
+	tg *konnectv1alpha1.KonnectCloudGatewayTransitGateway,
+) error {
+	networkID := tg.GetNetworkID()
+	if networkID == "" {
+		return CantPerformOperationWithoutNetworkIDError{
+			Entity: tg,
+			Op:     CreateOp,
+		}
+	}
+	// Failures on creating transit gateway causes `sdk.CreateTransitGateway` stuck because the SDK performs a retry that can take up to 1 hour:
+	// https://github.com/Kong/gateway-operator/issues/1521
+	// So we disable the retry here. The reconciler retries the creation by the requeue mechanism of the controller runtime.
```

Comment:
I think we can make these comments shorter (leaving the link to the issue). Simply saying: "We need to set the empty retry config to prevent using the default retry which causes failures and blocks the reconciliation". 

WDYT?

Reply:
done.